api/v1alpha1/databaseengine_types.go

```diff
@@ -75,6 +75,27 @@ type (
 type DatabaseEngineSpec struct {
 	Type            EngineType `json:"type"`
 	AllowedVersions []string   `json:"allowedVersions,omitempty"`
+	// SecretKeys contains the definition of the various Secrets that
+	// the given DBEngine supports.
+	// This information acts like metadata for the Everest UI to guide the users
+	// in filling out the correct Secret keys for their clusters.
+	// +optional
+	SecretKeys SecretKeys `json:"secretKeys,omitempty"`
+}
+
+// SecretKeys contains the definition of the various Secrets that
+// the given DBEngine supports.
+type SecretKeys struct {
+	// User secret keys are used to store the details of the users.
+	User []SecretKey `json:"user,omitempty"`
+}
+
```

Comment:
Have you considered to have `[]SecretKey` where each SecretKey contains the type (in this case, the type would be 'user') instead of the `SecretKeys` structure? Do you expect the structure to be different for different kinds of secrets?

Reply:
Please see section 8.4 of this doc - https://www.notion.so/percona/External-Backup-Imports-1ea674d091f38005bd5bc063a9d1862c?source=copy_link#1ed674d091f38050b996c9d8aff34dc8